thetis/shallowwater_eq.py

```diff
@@ -650,13 +650,23 @@ class QuadraticDragTerm(ShallowWaterMomentumTerm):
     def residual(self, uv, eta, uv_old, eta_old, fields, fields_old, bnd_conditions=None):
         total_h = self.depth.get_total_depth(eta_old)
         manning_drag_coefficient = fields_old.get('manning_drag_coefficient')
+        nikuradse_bed_roughness = fields_old.get('nikuradse_bed_roughness')
         C_D = fields_old.get('quadratic_drag_coefficient')
         f = 0
         if manning_drag_coefficient is not None:
             if C_D is not None:
                 raise Exception('Cannot set both dimensionless and Manning drag parameter')
             C_D = g_grav * manning_drag_coefficient**2 / total_h**(1./3.)
 
+        if nikuradse_bed_roughness is not None:
+            if manning_drag_coefficient is not None:
+                raise Exception('Cannot set both Nikuradse drag and Manning drag parameter')
+            if C_D is not None:
+                raise Exception('Cannot set both dimensionless and Nikuradse drag parameter')
+
+            kappa = physical_constants['von_karman']
+            C_D = conditional(total_h > nikuradse_bed_roughness, 2*(kappa**2)/(ln(11.036*total_h/nikuradse_bed_roughness)**2), Constant(0.0))
```

Comment:
I'm a little surprised by this conditional: I can see with `total_h<nikuradse_bed_roughness` you get in a different regime, but this totally switches the drag off which seems the opposite of what you want in a thin layer of water....

Reply:
I have checked and this agrees with the formulation in telemac/sisyphe (http://docs.opentelemac.org/tob__sisyphe_8f_source.html). As discussed in the call, I think this limit is put in because of wetting and drying. Given that nikuradse_bed_roughness is normally defined as 3 times the average sediment size (O(10^(-4)), this is very unlikely to be less than the wetting and drying alpha prescribed by the user